src/app/rundown-view/factories/key-binding.factory.ts

```diff
@@ -142,7 +145,17 @@ export class KeyBindingFactory {
     if (rundown.mode === RundownMode.ACTIVE) {
       return
     }
-    this.dialogService.createConfirmDialog(rundown.name, 'Are you sure you want to activate the Rundown?', 'Activate', () => this.rundownService.activate(rundown.id).subscribe())
+    const currentlyNonIdleBasicRundown: BasicRundown | undefined = this.basicRundownStateService.getNonInactiveRundown()
+    if (currentlyNonIdleBasicRundown) {
+      this.dialogService.createConfirmDialog(
+        rundown.name,
+        `Are you sure you want to activate the Rundown?\n\nThis will deactivate the Rundown "${currentlyNonIdleBasicRundown.name}"`,
+        'Activate',
+        () => this.rundownService.deactivate(currentlyNonIdleBasicRundown.id).subscribe(() => this.rundownService.activate(rundown.id).subscribe())
+      )
+    } else {
+      this.dialogService.createConfirmDialog(rundown.name, 'Are you sure you want to activate the Rundown?', 'Activate', () => this.rundownService.activate(rundown.id).subscribe())
+    }
```

Comment:
This is duplicated with the code in the RundownHeaderContextMenu.
Whenever you start writing duplicated code, something in you should trigger and you should try to see if it's possible to avoid it. In  this case, there is not really anything that is component specific, so the code could be extracted to a service. Just to give an example.